common/shamir-secret.c

```diff
@@ -0,0 +1,267 @@
+#include <stdio.h>
+#include </opt/homebrew/Cellar/gmp/6.2.1_1/include/gmp.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* 1 + 2^121 + 2^178 + 2^241 + 2^256 */
+const char *prime_polynomial_str = "115795622931445752192230534727574610162518887676182672686415539519216821469185";
```

Comment:
static const char prime_polynomial_str[] = "..."

static is always better (shows it's not used externally), and [] means it's literally the string, not a string and then we store the address in the variable.